### PR TITLE
📦 NEW: dark mode theme

### PR DESCRIPTION
**Pull request by rgvillanueva28**
#55

### CODE REVIEW

**Comment by warengonzaga** [comment]
Looks like has a few conflicts @rgvillanueva28 what branch did you use as base?

**Comment by warengonzaga** [comment]
I was about to fix the PR @rgvillanueva28 but thank you! Let us review it first.

**Comment by warengonzaga** [comment]
Amazing work @rgvillanueva28! I like it, can we make the dark mode default? Then we can adjust the colors from there. cc: @jnebab @kquirapas

**Comment by rgvillanueva28** [comment]
Yup. Dark mode by default is possible. Will update it soon.

**Review by warengonzaga — CHANGES_REQUESTED** [review]
Just make the dark mode default and we are happy to merge it to the v1.3.0 release. Thank you for your contribution! 💪

**Comment by kquirapas** [comment]
Will Review

**Review by warengonzaga — APPROVED** [review]
LGTM 💪 fantastic contribution @rgvillanueva28

**Comment by jnebab** [comment]
Thanks for the contrib. Just have one minor issue in mobile. The position of the switch button is off. Kindly fix it.

image

File: src/components/DarkModeSwitch.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 transform cursor-pointer transition-opacity duration-200";
+
+function DarkModeSwitch() {
+  //Set light theme by default
+  const [theme, setTheme] = useState("dark");
+
+  useEffect(() => {
+    //Set theme on device
+    if (typeof window !== "undefined") {
+      if (localStorage.theme === "dark" || !("theme" in localStorage)) {
```
Comment (jnebab) [inline src/components/DarkModeSwitch.tsx:14]: setting and saving the selected theme in local storage is ok, but we should be able to detect the current prefer of the user via matchmedia. You can have a look at this [article](https://blog.logrocket.com/dark-mode-in-react-an-in-depth-guide/) for some idea if not familiar with match media yet.

**Review by jnebab — CHANGES_REQUESTED** [review]
fix mobile screen issue of the switch button and use matchMedia API to detect user's system preference and use it as the default theme

File: src/components/DarkModeSwitch.tsx (hunk 1 above)
Comment (rgvillanueva28) [reply src/components/DarkModeSwitch.tsx:14]: I used matchmedia at first but Sir Warren asked to make dark mode as default [here].(https://github.com/web3phl/directory/pull/62#pullrequestreview-1096679545)
I will try to apply the [useLocalStorage](https://usehooks-ts.com/react-hook/use-local-storage) hook.
Thank you for the recommendations and inputs.

Comment (warengonzaga) [reply src/components/DarkModeSwitch.tsx:14]: Good catch @jnebab for this one!

**Comment by warengonzaga** [comment]
> fix mobile screen issue of the switch button and use matchMedia API to detect user's system preference and use it as the default theme

Actually agree with this we should follow the matchMedia API for user preference. Didn't think of it. Apologies @rgvillanueva28 🙏 I think that's our requirement for now. We can improve the colors once we merge this. cc: @jnebab

File: src/components/DarkModeSwitch.tsx (hunk 1 above)
Comment (warengonzaga) [reply src/components/DarkModeSwitch.tsx:14]: > I used matchmedia at first but Sir Warren asked to make dark mode as default [here].([#62 (review)](https://github.com/web3phl/directory/pull/62#pullrequestreview-1096679545)) I will try to apply the [useLocalStorage](https://usehooks-ts.com/react-hook/use-local-storage) hook. Thank you for the recommendations and inputs.

My bad, @jnebab has point, if the user's system prefers dark mode we should follow that instead. If not we should follow too.

Comment (jnebab) [reply src/components/DarkModeSwitch.tsx:14]: > I used matchmedia at first but Sir Warren asked to make dark mode as default [here].([#62 (review)](https://github.com/web3phl/directory/pull/62#pullrequestreview-1096679545)) I will try to apply the [useLocalStorage](https://usehooks-ts.com/react-hook/use-local-storage) hook. Thank you for the recommendations and inputs.

my apologies too, didn't read the previous comments before I made my review. if you can do the change, we will appreciate it very much. Thank you @rgvillanueva28

**Comment by jnebab** [comment]
> > fix mobile screen issue of the switch button and use matchMedia API to detect user's system preference and use it as the default theme
> 
> Actually agree with this we should follow the matchMedia API for user preference. Didn't think of it. Apologies @rgvillanueva28 🙏 I think that's our requirement for now. We can improve the colors once we merge this. cc: @jnebab

agree!

**Comment by warengonzaga** [comment]
Alright looks like it is all good, I'm happy to merge it! Thanks, @jnebab, and @rgvillanueva28 for the contribution!

**Pull request merged**